addon/components/flash-message.js

```diff
@@ -30,17 +30,20 @@ export default class FlashMessage extends Component {
   @bool('template')
   hasBlock;
 
-  @computed('flash.type', 'messageStyle')
+  @computed('messageStyle')
+  get messageStylePrefix() {
+    const isFoundation = this.messageStyle === 'foundation';
+    return isFoundation ? 'alert-box ' : 'alert alert-';
+  }
+
+  @computed('flash.type', 'flashTypePrefix', 'messageStylePrefix')
   get alertType() {
     const flashType = this.flash.type || '';
-    const messageStyle = this.messageStyle || '';
-    let prefix = 'alert alert-';
 
-    if (messageStyle === 'foundation') {
-      prefix = 'alert-box ';
-    }
+    const { flashTypePrefix, messageStylePrefix } = this;
```

Comment:
What do you thin about adding a warning for the developer if they have both of these things set on the component? i.e. if they have `messageStyle` defined but also set `flashTypePrefix` they might have an expectation that both will be honored.  We could add a simple console warning saying that that `messageStyle` isn't being enforced

Reply:
I did have this implemented, but then removed it, because `messageStyle` has a default value of `bootstrap` so it's always defined.

Reply:
🤦 Oh right